upstream_utils/upstream_utils.py

```diff
@@ -135,7 +136,22 @@ def comment_out_invalid_includes(filename, include_roots):
             f.write(new_contents)
 
 
-def git_am(patch, use_threeway=False, ignore_whitespace=False):
+def has_prefix(path: Path, prefix: Path):
+    """Checks if a path has a certain prefix.
+
+    Keyword arguments:
+    path -- path to check if it begins with the prefix
+    prefix -- prefix to use
+
+    Returns:
+    True if the path begins with the prefix, False otherwise.
+    """
+    return len(path.parts) >= len(prefix.parts) and True in [
+        path.parts[i] == prefix.parts[i] for i in range(len(prefix.parts))
+    ]
```

Comment:
If you give `zip()` lists of length 3 and 2, you'll get tuples for the first 2 elements.
```suggestion
    return len(path.parts) >= len(prefix.parts) and all([
        p1 == p2 for p1, p2 in zip(path.parts, prefix.parts)
    ])
```